### DISCUSSION TITLE
graphic don't show 'neighbour'lines

### DISCUSSION THREAD

**Discussion by Nifrabar** [zwave-js/zwave-js-ui#3599] [General]
Hi!
First Q here.
I have ZwaveJs-UI running since a few days.
Before I used OZW without much trouble.
One of the problems I am facing: graphics do show all nodes  but not the lines in between the nodes. Is this a setting?
How to make the lines visible?
Thanks for your help.
-Bart

**Comment by robertsLando** [comment]
@Nifrabar the "old" zwave graph you used to see on all applications were using only node neighbors informations, the graph z-ui show uses instead the routes provided by the controller so that is much more reliable then before, neighbors informations alone are useless and misleading based on our experience

**Reply by Nifrabar** [reply]
thanks for your reply. How can I manage to show the controller provided routings?

**Reply by robertsLando** [reply]
you see them in the zwave network graph. You can also customize priority/return routes. See https://zwave-js.github.io/zwave-js-ui/#/screenshots?id=mesh

**Reply by Nifrabar** [reply]
I am sorry to say but in graphics I only see the purple star being the controller and several black figures representing the nodes. Do I have to do settings to get graphic as per your example? 
Remember I'm new to ZwaveJs sk probably missing something.

**Reply by robertsLando** [reply]
@Nifrabar What z-ui version are you using ? It could be your devices are battery powered so I dunno the routes till you wake up them. BTW I kindof fixed this in one of the latest releases where I considered the nlwr in the overview

**Comment by Nifrabar** [comment]
@robertsLando
How to find the version?
Just one node is batterypowered
-Bart

**Reply by robertsLando** [reply]
check the `i` (info) icon in topbar on the right

**Comment by Nifrabar** [comment]
zwave-js-ui: 9.3.1.f26b79e

zwave-js: 12.2.3

home id: 25485431

home hex: 0x184e077

 

This is running in docker.

 

Van: Daniel Lando ***@***.***> 
Verzonden: dinsdag 13 februari 2024 08:23
Aan: zwave-js/zwave-js-ui ***@***.***>
CC: Nifrabar ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [zwave-js/zwave-js-ui] graphic don't show 'neighbour'lines (Discussion #3599)

 

check the i (info) icon in topbar on the right

**Reply by robertsLando** [reply]
That's a very old one. Please update to latest 9.8.3

**Comment by Nifrabar** [comment]
The dockercontainer which I used last week  following the instructions of Domoticz Wiki came up with this (old) version. From ZwaveJs-UI it's impossible to update. I don't know how I can update now.
https://www.domoticz.com/wiki/Zwave-JS-UI

**Reply by robertsLando** [reply]
docker-compose pull
docker-compose restart

**Reply by Nifrabar** [reply]
not recognised from raspberry pi command line
error:
-bash : docker-compose : opdracht niet gevonden
translated : command not found

**Reply by kpine** [reply]
```console
docker compose pull
docker compose up -d
```

**Reply by Nifrabar** [reply]
sorry , message after docker compose pull
no configuration file provided: not found

**Reply by Nifrabar** [reply]
Oke I managed but still same versions
Zwave Js Ui 9.3.1
Zwave Js 12.2.3

**Reply by kpine** [reply]
You need to run the command in the same directory you created the docker compose file, which was done in the installation step. https://www.domoticz.com/wiki/Zwave-JS-UI#Installing_ZWaveJS-UI

If you are seeing the same version, 9.3.1, the process was not followed correctly.

**Reply by Nifrabar** [reply]
At long last after consulting a linux expert I managed to get the updated app.
So now again heading towards the topic. No lines visible between all nodes in the graphic.

**Reply by kpine** [reply]
Not all controllers provide routing information. It's possible you won't see any information if that's the case. This does not necessarily indicate any problem.

Perhaps you can provide a bit more detail about your install because the information so far is limited.

- What controller are you using? What is the SDK version?
- Are the nodes w/o connecting lines battery, mains-powered, or both?
- Please provide a screenshot and node debug info of a problem node.
- Please enable [driver debug logs](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs) and attach a log of the driver startup process

**Reply by Nifrabar** [reply]
Sorry for late reply.
ZwaveJS running in docker-compose  (latest version)

zwave-js-ui: 9.8.3.3e632db
zwave-js: 12.4.3
home id: 25485431
home hex: 0x184e077

Installed on raspberry pi v3+ OS Bullseye

These are my devices.
![image](https://github.com/zwave-js/zwave-js-ui/assets/20126468/58b1cbed-b853-4dc2-a19e-4c7950c5c74c)
![image](https://github.com/zwave-js/zwave-js-ui/assets/20126468/ab820605-11e5-4983-a008-02696453985f)
![image](https://github.com/zwave-js/zwave-js-ui/assets/20126468/43ab5488-caf4-4f55-8ef2-bb2060118d05)
![image](https://github.com/zwave-js/zwave-js-ui/assets/20126468/4f879f5f-9224-496d-84cb-e6197be2bbab)

Frontend : Domoticz 2024.4 
Node 1 the controller DSA02203 AEON Labs / Z-Stick S2 USB Controller
Node 2 is a Neo Coolcam Motionsensor.
Node 4 is weird. Naming and Type differs from options. Probably leftover because I restarted some times form scratch
Node 9 is a Eurotronics / Thermostatic Valve 
Batteries are new
All other nodes are energy monitors Everspring AN158 and Everspring AN181

Node 2 cannot be included for unknown reason. Have reset the device and retried several times (BTW just for info and not to blame you: using OZW it was possible to include this device. Then I started from scratch. Tried to include with ZwaveJS without succes.

Node 9 once disappeared form the controlpanel out of the blue.
Node 9 control of thermostat is not always as I expect. Controlling (setpoint 1)from devices sometimes is refected in ZwaveJS and sometimes not at all.  Vica versa the same. Setpont 11 I don't understand what to do with that.

Domoticz cannot set or receive setpoint of node 9. I just tried with on/off node 11. That is OK in Domoticz although some sec delay..

Q1 is there a way to include node 2?
Q2 how can I delete the dead node 4 ? tried th regular way but controller keeps saying "exclusion stopped"
Q3 what can I do to get proper interface between domoticz and node9 vice versa setpoint 1.
Q4 how can I make visible the routing in the graphic. Just as an example I included the screenshot of node 6. 
Possibly I need to do some settings somewhere?

How do I provide you with the mentioned logs? 
 
Thanks for your help and advices.
-Bart

**Reply by robertsLando** [reply]
IMO Z-Stick S2 is just too old and doesn't provides routing info...

**Reply by Nifrabar** [reply]
@RobertsLando
Thanks for reply. What controller is adviced.
Might an other controller solve other mentioned problems?

**Comment by robertsLando** [comment]
@Nifrabar could you attach here an export of your nodes please? Just press on the bottom right fab on control panel, click then the purple button, and then on dump:

![2024-02-22_13-35](https://github.com/zwave-js/zwave-js-ui/assets/11502495/b4328aeb-4552-47b5-9a77-526fb50aa1cf)

This will show me if any of your node do have some routes or not. Then About your other problems:

> Q1 is there a way to include node 2?

Only way to debug such issues is to make a [driver log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs), loglevel `debug` and attach it here as a file (drag & drop into the text field). Dunno if this is an issue related to the old stick or the device itself. Maybe try to factory reset the device and try to include it again.

> Q2 how can I delete the dead node 4 ? tried th regular way but controller keeps saying "exclusion stopped"

If the node is dead and you don't have access to that device anymore the only way is to call the remove failed node action against it. Otherwise you have to put the dead device in exclusion mode before enabling exclusion on your controller

> Q3 what can I do to get proper interface between domoticz and node9 vice versa setpoint 1.

Maybe I can answer to this when I have nodes debug export. I will be able to see the discovery payload sent to Domoticz, maybe the setpoint topic is wrong for some reason

> Q4 how can I make visible the routing in the graphic. Just as an example I included the screenshot of node 6.

I think this is due to your old stick but I dunno

**Comment by Nifrabar** [comment]
[nodes_dump.json](https://github.com/zwave-js/zwave-js-ui/files/14375052/nodes_dump.json)

how to? driver log is over 26 mB

**Reply by robertsLando** [reply]
none of your nodes have `nlwr` or `lwr` routes so I think this is a controller limitation.

About the climate device I don't see any issue, try to ask in domotics forum/github issues

**Comment by Nifrabar** [comment]
OK unzipped. 
[zwavejs_2024-02-18.log.gz](https://github.com/zwave-js/zwave-js-ui/files/14375064/zwavejs_2024-02-18.log.gz)

**Reply by robertsLando** [reply]
cc @AlCalzone

**Comment by Nifrabar** [comment]
[zwavejs_2024-02-22.log](https://github.com/zwave-js/zwave-js-ui/files/14375087/zwavejs_2024-02-22.log)

**Comment by Nifrabar** [comment]
thanks fof fast repliesany suggestion for new zwave controller? Verzonden vanaf Outlook voor Android

**Reply by robertsLando** [reply]
Zooz ZSt39
